cloudferrylib/os/image/glance_image.py

```diff
@@ -36,26 +40,56 @@ class GlanceImage(image.Image):
     The main class for working with Openstack Glance Image Service.
 
     """
+
     def __init__(self, config, cloud):
         self.config = config
         self.host = config.cloud.host
         self.cloud = cloud
         self.identity_client = cloud.resources['identity']
+        # get mysql settings
+        self.config = config
         self.glance_client = self.proxy(self.get_client(), config)
         super(GlanceImage, self).__init__(config)
 
-    def get_client(self):
+    def get_db_connection(self):
+        if not hasattr(
+                self.cloud.config,
+                self.cloud.position + '_image'):
+            LOG.debug('running on default mysql settings')
+            return mysql_connector.MysqlConnector(
+                self.config.mysql, 'glance')
+        else:
+            LOG.debug('running on custom mysql settings')
+            my_settings = getattr(
```

Comment:
IRL we have mysql databases on different hosts - we need to specify endpoint for each component